### ISSUE TITLE
KL_divergence subtraction z_dim

### ISSUE THREAD

**Issue by lw3259111** [paruby/FactorVAE#1]
`
KL_divergence = 0.5 * tf.reduce_mean(tf.reduce_sum(tf.exp(self.enc_logvar) - self.enc_logvar + self.enc_mean**2,axis=1) - self.z_dim)
`
Thanks for your code, but i meet a problem for your code, In line 137, why need you substraction z_dim?
In origin pytorch code, i can't found the operator.

**Comment by abdulfatir** [comment]
In the KL expression you have `-1` and you take a sum over the latent dimension. This would just sum to `-z_dim` which is what is written.
